lib/internal/modules/esm/fetchModule.js

```diff
@@ -0,0 +1,287 @@
+'use strict';
+const {
+  globalThis,
+  ArrayPrototypePush,
+  Promise,
+  PromisePrototypeThen,
+  PromiseResolve,
+  SafeMap,
+  StringPrototypeEndsWith,
+  StringPrototypeSlice,
+  StringPrototypeStartsWith,
+} = primordials;
+const {
+  Buffer: {
+    concat: BufferConcat
+  }
+} = globalThis;
+const {
+  ERR_NETWORK_IMPORT_DISALLOWED,
+  ERR_NETWORK_IMPORT_BAD_RESPONSE,
+} = require('internal/errors').codes;
+const { URL } = require('internal/url');
+const net = require('net');
+const path = require('path');
+
+/**
+ * @typedef CacheEntry
+ * @property {Promise<string> | string} resolvedHREF
+ * @property {Record<string, string>} headers
+ * @property {Promise<Buffer> | Buffer} body
+ */
+
+/**
+ * Only for GET requests, other requests would need new Map
+ * HTTP cache semantics keep diff caches
+ *
+ * Maps HREF to pending cache entry
+ * @type {Map<string, Promise<CacheEntry> | CacheEntry>}
+ */
+const cacheForGET = new SafeMap();
+
+
+/**
+ * Lazily load these to avoid snapshot issues
+ */
+let HTTPSAgent;
+function HTTPSGet(url, opts) {
+  const https = require('https');
+  HTTPSAgent ??= new https.Agent({
+    keepAlive: true
+  });
+  return https.get(url, {
+    agent: HTTPSAgent,
+    ...opts
+  });
+}
+
+let HTTPAgent;
+function HTTPGet(url, opts) {
+  const http = require('http');
+  HTTPAgent ??= new http.Agent({
+    keepAlive: true
+  });
+  return http.get(url, {
+    agent: HTTPAgent,
+    ...opts
+  });
```

Comment:
Is there something about HTTPS imports that requires `keepAlive: true`? If not, we should just use the default global agent (what you get via `require('https').globalAgent`, but it’s the default so it doesn’t need to be specified):

```suggestion
  const https = require('https');
  return https.get(url, { …opts });
}

let HTTPAgent;
function HTTPGet(url, opts) {
  const http = require('http');
  return http.get(url, { …opts });
```

If we need `keepAlive: true`, I think we should make parallel global agents for network imports (like `require('https').globalNetworkImportsAgent`). The purpose of the global agents is to allow users to override Node’s defaults for network requests, like to set up an HTTP proxy or custom DNS resolution (or enable `keepAlive` all the time, for example).

Reply:
do not use the global agent, this is somewhere in the comments, but this is partially a perf improvement but also to avoid someone tainting the agent

Reply:
if we want to setup HTTP proxy/DNS it shouldn't be through the Reflective API since it cannot mutate prior to `import`

Reply:
Okay then we should add a comment here, where we define `HTTPSAgent` etc., explaining that we’re specifically _not_ using `globalAgent` and why.